src/main/java/net/pms/dlna/DLNAImage.java

```diff
@@ -0,0 +1,527 @@
+import net.pms.util.ImagesUtil.ScaleType;
+
+/**
+ * This class is simply a byte array for holding an {@link ImageIO} supported
+ * image with some additional metadata restricted to valid DLNA image media
+ * format profiles.
+ *
+ * @see DLNAThumbnail
+ *
+ * @author Nadahar
+ */
+public class DLNAImage extends Image {
+
+	private static final long serialVersionUID = 1L;
+	protected final DLNAImageProfile profile;
+
+	/**
+	 * Creates a new {@link DLNAImage} instance.
+	 *
+	 * @param bytes the source image in either GIF, JPEG or PNG format adhering
+	 * to the DLNA restrictions for color space and compression.
+	 * @param imageInfo the {@link ImageInfo} to store with this
+	 *                  {@link DLNAImage}.
+     * @param profile the {@link DLNAImageProfile} this {@link DLNAImage}
+     *                adheres to.
+	 * @param copy whether this instance should be copied or shared.
+	 */
+	public DLNAImage(
+		byte[] bytes,
+		ImageInfo imageInfo,
+		DLNAImageProfile profile,
+		boolean copy
+	) {
+		super(bytes, imageInfo, copy);
+		this.profile = profile != null ? profile : findMatchingProfile(this instanceof DLNAThumbnail);
+		if (
+			this.profile == null ||
+			!this.profile.checkCompliance(imageInfo).isAllCorrect()
+		) {
+			throw new IllegalArgumentException("DLNAImage: Compliance check failed");
+		}
+	}
+
+	/**
+	 * Creates a new {@link DLNAImage} instance.
+	 *
+	 * @param bytes the source image in either GIF, JPEG or PNG format adhering
+	 * to the DLNA restrictions for color space and compression.
+	 * @param width the width of the source image.
+	 * @param height the height of the source image.
+	 * @param format the {@link ImageFormat} of the source image.
+	 * @param colorModel the {@link ColorModel} of the image.
+	 * @param metadata the {@link Metadata} instance describing the image.
+     * @param profile the {@link DLNAImageProfile} this {@link DLNAImage}
+     *                adheres to.
+	 * @param copy whether this instance should be copied or shared.
+	 */
+	public DLNAImage(
+		byte[] bytes,
+		int width,
+		int height,
+		ImageFormat format,
+		ColorModel colorModel,
+		Metadata metadata,
+		DLNAImageProfile profile,
+		boolean copy
+	) {
+		super(bytes, width, height, format, colorModel, metadata, true, copy);
+		this.profile = profile != null ? profile : findMatchingProfile(this instanceof DLNAThumbnail);
+		if (
+			this.profile == null ||
+			!this.profile.checkCompliance(imageInfo).isAllCorrect()
+		) {
+			throw new IllegalArgumentException("DLNAImage: Compliance check failed");
+		}
+	}
+
+	/**
+	 * Creates a new {@link DLNAImage} instance.
+	 *
+	 * @param inputByteArray the source image in either GIF, JPEG or PNG format
+	 * adhering to the DLNA restrictions for color space and compression.
+	 * @param format the {@link ImageFormat} the source image is in.
+	 * @param bufferedImage the {@link BufferedImage} to get
+	 *                      non-{@link Metadata} metadata from.
```

Comment:
That's what the link and the capitalization is there for; to create a distinction. ```Metadata``` is the name of a class, ```com.drew.metadata.Metadata``` in full, while "metadata" is just the regular word meaning data that provides information about other data. The ```Metadata``` class holds metadata, I'm referring to those metadata NOT helt by ```Metadata```. I know it's complicated, but how should I put it to make it easy?